keyboards/cipulot/common/eeprom_tools.h

```diff
@@ -13,24 +13,14 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-
 #pragma once
 
-#include <stdint.h>
-#include <stdbool.h>
-
-#include "matrix.h"
-
-typedef struct {
-    uint16_t ecsm_actuation_threshold; // threshold for key release
-    uint16_t ecsm_release_threshold;   // threshold for key press
-} ecsm_config_t;
+#include "eeprom.h"
 
-ecsm_config_t ecsm_config;
+#if (EECONFIG_KB_DATA_SIZE) > 0
+#    define EEPROM_KB_PARTIAL_UPDATE(__struct, __field) eeprom_update_block(&(__struct.__field), (void *)((void *)(EECONFIG_KB_DATABLOCK) + offsetof(typeof(__struct), __field)), sizeof(__struct.__field))
+#endif
 
-int      ecsm_init(ecsm_config_t const* const ecsm_config);
-int      ecsm_update(ecsm_config_t const* const ecsm_config);
-bool     ecsm_matrix_scan(matrix_row_t current_matrix[]);
-uint16_t ecsm_readkey_raw(uint8_t channel, uint8_t row, uint8_t col);
-bool     ecsm_update_key(matrix_row_t* current_row, uint8_t row, uint8_t col, uint16_t sw_value);
-void     ecsm_print_matrix(void);
+#if (EECONFIG_USER_DATA_SIZE) > 0
+#    define EEPROM_USER_PARTIAL_UPDATE(__struct, __field) eeprom_update_block(&(__struct.__field), (void *)((void *)(EECONFIG_USER_DATABLOCK) + offsetof(typeof(__struct), __field)), sizeof(__struct.__field))
```

Comment:
Not being used (?)

Reply:
This is for future usage, in case I ever need to have the `USER` version of the function. I included it in the same tools file to have it ready when needed.